servers/services/src/main/java/org/projectnessie/services/rest/ContentsResource.java

```diff
@@ -60,9 +63,10 @@ protected SecurityContext getSecurityContext() {
   @Override
   public Contents getContents(ContentsKey key, String namedRef, String hashOnRef)
       throws NessieNotFoundException {
-    Hash ref = namedRefWithHashOrThrow(namedRef, hashOnRef).getHash();
+    WithHash<NamedRef> ref = namedRefWithHashOrThrow(namedRef, hashOnRef);
+    getAccessChecker().canReadEntityValue(createAccessContext(), ref.getValue(), key);
```

Comment:
Is it not possible to add this as a filter or something?

Reply:
you mean like a HTTP filter?

Reply:
yes, a jax-rs filter. I guess not easily as each  call to the access checker is slightly different. It just feels a bit icky to have the authz logic embedded in the operational logic

Reply:
It should be possible to abstract authz checking from the actual implementation.
Either by delegation or "naive `public class AuthzContentsResource extends ContentsResource`".

It feels, that it should also be possible to have a separate module `nessie-authz` next to `servers/services` with the "right" CDI annotations, so a `AuthzContentsResource` + `AuthzTreeResource` are preferred by the container over the "raw" `ContentsResource` + `TreeResource`.

Reply:
I agree it should definitely be outside of the operational logic. My first goal was to have a prototype so that we can decide whether we want to configure authz rules via a quarkus config

Reply:
I experimented a bit with Jax-RS filters, but this really makes things quite difficult as for some checks (e.g. when creating a branch/tag, which is a `POST`) as we need to access the `POST` body and such in order to perform our checks. Thus I figured it's easier to have a `TreeResourceWithAuthorizationChecks` class that performs those checks and then just delegates to `TreeResource` for the normal functionality.